pretty_midi/fluidsynth.py

```diff
@@ -53,10 +55,14 @@ def get_fluidsynth_instance(synthesizer=None, sfid=0, fs=44100):
         sf2_path = synthesizer
         if not os.path.exists(synthesizer):
             raise ValueError("No soundfont file found at the supplied path {}".format(sf2_path))
+        fs = fs or DEFAULT_SAMPLE_RATE
         synthesizer = fluidsynth.Synth(samplerate=fs)
         sfid = synthesizer.sfload(sf2_path)
         new_instance_created = True
     elif isinstance(synthesizer, fluidsynth.Synth):
+        synth_fs = synthesizer.get_setting('synth.sample-rate')
+        if fs and synth_fs != fs:
+            raise ValueError("synth sample-rate does not match fs", synth_fs)
```

Comment:
May as well make the error message more helpful `f"synthesizer sample rate of {synth_fs} doesn't match provided fs of {fs}"`